### PR TITLE
gepifanio/c01-git03

### PR DESCRIPTION
**Pull request by gepifanio**
# Description

Exercise (e.g. C01-E01): c01-git03 

Student Name: Guilherme Oliveira Epifanio

> A brief explanation of what has been done.

Git history testing with previous big files.

## Type of change

- [x] Exercise Submission (one exercise per PR)
  - [x] This PR is just for **one** exercise of a class
  - [x] PR name follows the pattern `<github-username>/<exercise-number>`
  - [x] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-git03/gepifanio/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+
+git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch path_to_file" HEAD
+
+
+If you want to push it to remote repo just do git push
+
+git push -all
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/gepifanio/README.md:80]: As it is rewriting , you'll need `git push --all --force` here

File: classes/01class/exercises/c01-git03/gepifanio/README.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+1. What are the consequences of the previous action?
+
+It rewrites the whole history, which changes all the commit ids and will definately affect other users of the same repository.
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/gepifanio/README.md:84]: Hi Gui, your answer looks good. 

The `filter-branch` command will rewrite the commit where the file indicated is modified and any other commits in branches that depend on it somehow.

Comment (denstorti) [reply classes/01class/exercises/c01-git03/gepifanio/README.md:84]: How's it is going to impact other users?

File: classes/01class/exercises/c01-git03/gepifanio/README.md (hunk 1 above)
Comment (gepifanio) [reply classes/01class/exercises/c01-git03/gepifanio/README.md:80]: Cheers

File: classes/01class/exercises/c01-git03/gepifanio/README.md (hunk 2 above)
Comment (gepifanio) [reply classes/01class/exercises/c01-git03/gepifanio/README.md:84]: Fixed ;) Cheers

**Pull request merged**